lib/srv/alpnproxy/kube.go

```diff
@@ -37,18 +50,34 @@ func (c KubeClientCerts) Add(teleportCluster, kubeCluster string, cert tls.Certi
 	c[common.KubeLocalProxySNI(teleportCluster, kubeCluster)] = cert
 }
 
+// KubeCertReissuer reissues a client certificate for a Kubernetes cluster.
+type KubeCertReissuer = func(ctx context.Context, teleportCluster, kubeCluster string) (tls.Certificate, error)
+
 // KubeMiddleware is a LocalProxyHTTPMiddleware for handling Kubernetes
 // requests.
 type KubeMiddleware struct {
 	DefaultLocalProxyHTTPMiddleware
 
+	// certReissuer is used to reissue a client certificate for a Kubernetes cluster if existing cert expired.
+	certReissuer KubeCertReissuer
+	// Clock specifies the time provider. Will be used to override the time anchor
+	// for TLS certificate verification.
+	// Defaults to real clock if unspecified
+	clock clockwork.Clock
+
+	certsMu sync.RWMutex
+	// certs is a map by cluster name of Kubernetes client certs.
 	certs KubeClientCerts
+
+	isCertReissuingRunning atomic.Bool
```

Comment:
add a comment to say this is used to limit the number of logins with required interaction to 1